server/caas/api.js

```diff
@@ -0,0 +1,324 @@
+        } else {
+          res.status(500).send({error: 'Invalid Token'});
+        }
+      }
+    });
+  });
+
+  app.use('/reset-password', function(req, res, next) {
+    var user = req.body;
+    var userEmail = user.email;
+    var name = user.name;
+    var url = config.apiServer + config.apiEndpoint + '/credentials?kind=password&publicValue=' + userEmail + '&limit=-1&sort=name';
+
+    newRequest({
+      url: url,
+      method: 'GET',
+    }, function(response, body) {
+
+      if (body.data.length) {
+
+        var credId = body.data[0].id;
+        var url = config.apiServer + config.apiEndpoint + '/accounts/' + body.data[0].accountId;
+
+        newRequest({
+          url: url,
+          method: 'GET'
+        }, function(response, body) {
+
+          if (body.type === 'account') {
+            generateAndInsertToken(body.id, name, userEmail, "reset", function(error, results, fields, token) {
+              if (error) {
+                res.status(500).send({error: error});
+              } else {
+
+                sendPasswordReset(req.body.email, response.body.name, req.headers.origin, token, function(error, resp) {
+                  if (error) {
+                    res.status(500).send({error: error});
+                  } else {
+                    res.status(200).json({success: 'Email sent'});
+                  }
+                });
+              }
+            });
+          } else {
+            res.status(500).send({error: 'Account not found'});
+          }
+        });
+      } else {
+        res.status(500).send({error: 'Credential not found'});
+      }
+    });
+  });
+
+  app.use('/update-password', function(req, res, next) {
+    var user = req.body;
+    var url = config.apiServer + config.apiEndpoint;
+
+    pool.query('SELECT * FROM challenge WHERE token = ? AND created > DATE_SUB(NOW(), INTERVAL 24 HOUR)', [user.token], function(error, results, fields) {
+      if (error) res.status(500).send({error: 'There was an error finding the record'});
+
+      if (results.length) {
+        var credential = results[0];
+
+        newRequest({
+          url: url + '/passwords?publicValue=' + credential.email,
+          method: 'GET'
+        }, function(response, body) {
+
+          newRequest({
+            url: url + '/passwords/' + body.data[0].id + '?action=changesecret',
+            method: 'POST',
+            body: {newSecret: user.pw, oldSecret: ''}
+          }, function(response, body) {
+            removeUserFromTable(credential.email, function(results, fields) {
+              getTokenForLogin(credential.email, user.pw, function(response, body) {
+                sendPasswordVerificationEmail(credential.email, credential.name, req.headers.origin, function(error, resp) {
+                  if (error) {
+                    res.status(500).send({error: 'error emailing verification'});
+                  } else {
+                    res.cookie('token', body.jwt, {secure: true});
```

Comment:
secure: req.secure, consider combining with the same thing earlier so there's one piece of code responsible for making the user logged in.